kiali/kiali
# Added support of Mesh config defaultExportTo fields.

### Describe the change
Mesh config has defaultExportTo values for DR, VS and SE (including Services) which should be considered when Istio config object does not have 'ExportTo' fields.

### Steps to test the PR

Make sure Istio's Global MeshConfig has values set for `defaultServiceExportTo`, `defaultVirtualServiceExportTo` and `defaultDestinationRuleExportTo`, for instance to `.`. By default the value is `*`.
![Screenshot from 2024-08-28 09-36-48](https://github.com/user-attachments/assets/3ca1f7cf-4bdc-4875-8ee2-4607e5254153)

Promote VS, DR and SE validation scenarios. On those objects which does not have 'ExportTo' themselves. Verify that these objects behave like they are not available in other namespaces.
Also verify that when objects have 'ExportTo' to `*`, it overrides the default `.`

Original issue scenario with SE in graph page should also work: https://github.com/kiali/kiali/issues/7590
Was not able to find the steps, will ask the issue owner to check.

### Automation testing

Existing unit tests modified, added more ones.

### Issue reference

https://github.com/kiali/kiali/issues/7590

Please note that we have an issue in validations which makes inconsistency between list and details pages when 'exportTo' field exists: https://github.com/kiali/kiali/issues/7685
